crates/scmi/README.md

```diff
@@ -0,0 +1,80 @@
+# vhost-user-scmi
+
+This program is a vhost-user backend for a VirtIO SCMI device.
+It provides SCMI access to various entities on the host; not
+necessarily only those providing an SCMI interface themselves.
+
+It is tested with QEMU's `-device vhost-user-scmi-pci` but should work
+with any virtual machine monitor (VMM) that supports vhost-user. See
+the Examples section below.
+
+## Synopsis
+
+**vhost-device-scmi** [*OPTIONS*]
+
+## Options
+
+.. program:: vhost-device-scmi
+
+.. option:: -h, --help
+
+  Print help.
+
+.. option:: -s, --socket-path=PATH
+
+  Location of the vhost-user Unix domain sockets.
+
+.. option:: -d, --device=SPEC
+
+  SCMI device specification in the format `ID,PROPERTY=VALUE,...`.
+  Can be used multiple times for multiple exposed devices.
+  Use `help` as the device ID to list help on all the available devices.
+
+You can set `RUST_LOG` environment variable to `debug` to get maximum
+messages on the standard error output.
+
+## Examples
+
+The daemon should be started first:
+
+::
+
+  host# vhost-device-scmi --socket-path=scmi.sock --device fake,name=foo
+
+The QEMU invocation needs to create a chardev socket the device can
+use to communicate as well as share the guests memory over a memfd:
+
+::
+
+  host# qemu-system \
+      -chardev socket,path=scmi.sock,id=scmi \
+      -device vhost-user-scmi-pci,chardev=vscmi,id=scmi \
+      -m 4096 \
+      -object memory-backend-file,id=mem,size=4G,mem-path=/dev/shm,share=on \
+      -numa node,memdev=mem \
```

Comment:
I would suggest using the `-machine` option to set the memory backend in the example, because `-numa` can not be available for some machine types:

`-machine <machine options>,memory-backend=mem`

Reply:
The same thing happened to me in the vsock device, where a user couldn't run the example on MIPS IIRC: https://github.com/rust-vmm/vhost-device/pull/253

Reply:
Thank you for this tip! Applied.